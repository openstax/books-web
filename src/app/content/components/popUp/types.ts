import { LinkedArchiveTree, LinkedArchiveTreeNode, LinkedArchiveTreeSection } from '../../types';

export type LocationFilters = Map<
  string,
  { section: LinkedArchiveTreeNode, children?: LinkedArchiveTreeSection[] }
>;

export type LocationFiltersWithChildren = Map<
  string,
  { section: LinkedArchiveTree, children: LinkedArchiveTreeSection[] }
>;

export interface FiltersChange<T> {
  remove: T[];
  new: T[];
}
