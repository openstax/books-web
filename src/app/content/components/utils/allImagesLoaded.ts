import { Document,  Element } from '@openstax/types/lib.dom';

export default (container: Element | Document) => {
  return Promise.all(Array.from(container.querySelectorAll('img')).map((img) => img.complete
    ? Promise.resolve()
    : new Promise<void>((resolve) => {

      const previousOnload = img.onload;
      const previousOnerror = img.onerror;

      img.onload = (...args) => {
        resolve();
        if (previousOnload) {
          previousOnload.apply(img, args);
        }
      };
      img.onerror = (...args) => {
        resolve();
        if (previousOnerror) {
          previousOnerror.apply(img, args);
        }
      };
    })
  )).then((): void => undefined);
};
