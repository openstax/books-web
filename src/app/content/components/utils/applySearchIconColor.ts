import { css } from 'styled-components/macro';
import theme from '../../../theme';
import { SearchButtonColor } from '../../types';
import { toolbarIconColor } from '../constants';

export const applySearchIconColor = (props: {colorSchema: SearchButtonColor}) =>
  props.colorSchema ? css`
    color: ${theme.color.primary[props.colorSchema].foreground};

    :hover,
    :focus {
        color: ${theme.color.primary[props.colorSchema].foreground};

        ::before {
            content:"";
            display: block;
            height: 100%;
            position: absolute;
            top: 0;
            right: 0;
            min-width: 45px;
            background-color: rgba(0, 0, 0, 0.06)
            ;
        }
    }
  ` : css`
  color: ${toolbarIconColor.base};
`;
