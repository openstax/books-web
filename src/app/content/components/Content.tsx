import React from 'react';
import styled, { css } from 'styled-components/macro';
import Layout from '../../components/Layout';
import ScrollOffset from '../../components/ScrollOffset';
import ErrorBoundary from '../../errors/components/ErrorBoundary';
import Notifications from '../../notifications/components/Notifications';
import theme from '../../theme';
import Footer from './../../components/Footer';
import Attribution from './Attribution';
import BookBanner from './BookBanner';
import CenteredContent from './CenteredContent';
import {
  bookBannerDesktopMiniHeight,
  bookBannerMobileMiniHeight,
  contentWrapperMaxWidth,
  mainContentBackground,
  sidebarDesktopWidth,
  sidebarTransitionTime,
  toolbarDesktopHeight,
  toolbarMobileHeight
} from './constants';
import ContentPane from './ContentPane';
import Page from './Page';
import PrevNextBar from './PrevNextBar';
import Sidebar from './Sidebar';
import Toolbar from './Toolbar';
import { isOpenConnector, styleWhenSidebarClosed } from './utils/sidebar';
import Wrapper from './Wrapper';
import { wrapperPadding } from './Wrapper';

// tslint:disable-next-line:variable-name
const Background = styled.div`
  @media screen {
  overflow: visible; /* so sidebar position: sticky works */
  background-color: ${theme.color.neutral.darker};
  width: 100%;
  height: 100%;
  }
`;

// tslint:disable-next-line:variable-name
const ContentNotifications = styled(Notifications)`
  z-index: 1; /* above content */
  top: ${bookBannerDesktopMiniHeight + toolbarDesktopHeight}rem;
  ${theme.breakpoints.mobile(css`
top: ${bookBannerMobileMiniHeight + toolbarMobileHeight}rem;
`)}
`;

// tslint:disable-next-line:variable-name
const CenteredContentRow = styled(CenteredContent)`
  @media screen {
    min-height: 100%;
    display: flex;
    flex-direction: row;
  }
`;

// tslint:disable-next-line:variable-name
const UndoPadding = isOpenConnector(styled.div`
  @media screen {
  overflow: visible;
  min-height: 100%;
  display: flex;
  flex-direction: column;
  margin-right: -${theme.padding.page.desktop}rem;
  ${theme.breakpoints.mobile(css`
  margin: 0 -${theme.padding.page.mobile}rem;
  `)}

    ${styleWhenSidebarClosed(css`
  margin-left: -${theme.padding.page.desktop}rem;
  ${theme.breakpoints.mobile(css`
        margin-left: -${theme.padding.page.mobile}rem;
  `)}
  `)}
  }
  `);

// tslint:disable-next-line:variable-name
const MainContentWrapper = isOpenConnector(styled.div`
  @media screen {
  flex: 1;
  display: flex;
  flex-direction: column;
  overflow: visible;
  background-color: ${mainContentBackground};
  transition: max-width ${sidebarTransitionTime}ms;
  width: 100%;
  max-width: ${contentWrapperMaxWidth - sidebarDesktopWidth}rem;
  ${theme.breakpoints.mobile(css`
  max-width: ${contentWrapperMaxWidth}rem;
  `)}
    ${styleWhenSidebarClosed(css`
  max-width: ${contentWrapperMaxWidth}rem;
  margin: 0 auto;
  `)}
  }
  `);

// tslint:disable-next-line:variable-name
const HideOverflowAndRedoPadding = isOpenConnector(styled.div`
  @media screen {
  flex: 1;
  ${wrapperPadding}
  ${styleWhenSidebarClosed(css`
  ${wrapperPadding}
  `)}
  }
  `);

/*
 * this layout is a mess for these reasons:
 * - the navs must have the default padding inside their containers so their
 *   backgrounds go to the edge of the window.
 *
 * - the content wrapper must behave the same way as the navs in order to get
 *   the sidebar in the right place to line up with the button in the toolbar.
 *
 * - the white background must then have negative margin to undo the default
 *   padding so that it can get to the edge of the screen on small windows.
 *
 * - the white background can't be on the whole wrapper, because it is fixed
 *   width on large screens, and then it wouldn't go to the edge on small ones.
 *
 * - transitioning the white background from full width to fixed width, while
 *   matching the container boundaries of the navbars, is complicated.
 *
 * - the default padding is duplicated inside the white margin for small
 *   screen behavior, but it cant affect the notifications or attribution,
 *   so there is another container for that.
 *
 * - the extra container for the padding inside the white margin can't go away
 *   because it is necessary to hide the overflow of items on the Page, and contain
 *   the margins of the Page, which must be overflow: visible.
 *
 * - when the sidebar is closed the white wrapper behaves more or less the same
 *   as the default wrapper, but when the sidebar is open it only needs
 *   padding/margin on the right, because the sidebar already puts the left
 *   side in the right place, and you don't want a gap between the sidebar
 *   and the content.
 *
 * - a bunch of these containers could be combined, except that then the
 *   transitions break because you're combining things that need to be
 *   fixed with things that need to be animated.
 *
 * - the whole layout depends on using max-width to resolve when the
 *   margins should and should not be applied, and changing margins
 *   all over the place, all of which means that a disturbing number
 *   of things need to know when the sidebar is open/closed.
 */
// tslint:disable-next-line:variable-name
const Content: React.SFC = () => <Layout>
  <ScrollOffset
    desktopOffset={bookBannerDesktopMiniHeight + toolbarDesktopHeight}
    mobileOffset={bookBannerMobileMiniHeight + toolbarMobileHeight}
  />
  <Background>
    <BookBanner />
    <Toolbar />
    <Wrapper>
      <ErrorBoundary>
        <CenteredContentRow>
          <Sidebar />
          <ContentPane>
            <UndoPadding>
              <MainContentWrapper>
                <ContentNotifications />
                <HideOverflowAndRedoPadding>
                  <Page />
                  <PrevNextBar />
                </HideOverflowAndRedoPadding>
                <Attribution />
                <Footer />
              </MainContentWrapper>
            </UndoPadding>
          </ContentPane>
        </CenteredContentRow>
      </ErrorBoundary>
    </Wrapper>
  </Background>
</Layout >;



export default Content;
