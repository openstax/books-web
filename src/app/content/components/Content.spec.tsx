import React from 'react';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import createTestStore from '../../../test/createTestStore';
import mockArchiveLoader, { book, shortPage } from '../../../test/mocks/archiveLoader';
import { mockCmsBook } from '../../../test/mocks/osWebLoader';
import MobileScrollLock from '../../components/MobileScrollLock';
import * as Services from '../../context/Services';
import MessageProvider from '../../MessageProvider';
import { AppServices, AppState, Store } from '../../types';
import { openToc } from '../actions';
import { Book } from '../types';
import { formatBookData } from '../utils';
import Content from './Content';
import { Sidebar } from './Sidebar';

describe('content', () => {
  let archiveLoader: ReturnType<typeof mockArchiveLoader>;
  let state: AppState;
  let store: Store;
  const services = {} as AppServices;
  const bookState: Book = formatBookData(book, mockCmsBook);

  beforeEach(() => {
    store = createTestStore({navigation: new URL('https://localhost/books/book-slug-1/pages/doesnotmatter')});
    state = store.getState();

    archiveLoader = mockArchiveLoader();
    (services as any).archiveLoader = archiveLoader;
  });

  it('matches snapshot', () => {
    state.content.book = bookState;
    state.content.page = shortPage;

    const component = renderer.create(<Provider store={store}>
      <Services.Provider value={services}>
        <MessageProvider>
          <Content />
        </MessageProvider>
      </Services.Provider>
    </Provider>);

    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders empty state', () => {
    const component = renderer.create(<Provider store={store}>
      <Services.Provider value={services}>
        <MessageProvider>
          <Content />
        </MessageProvider>
      </Services.Provider>
    </Provider>);

    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('gets page content out of cached archive query', () => {
    state.content.book = bookState;
    state.content.page = shortPage;

    renderer.create(<Provider store={store}>
      <Services.Provider value={services}>
        <MessageProvider>
          <Content />
        </MessageProvider>
      </Services.Provider>
    </Provider>);

    expect(archiveLoader.mock.cachedPage).toHaveBeenCalledTimes(1);
    expect(archiveLoader.mock.cachedPage).toHaveBeenCalledWith('testbook1-uuid', '1.0', 'testbook1-testpage4-uuid');
  });

  it('page element is still rendered if archive content is unavailable', () => {
    state.content.book = bookState;
    state.content.page = shortPage;

    archiveLoader.mock.cachedPage.mockReturnValue(undefined);

    const component = renderer.create(<Provider store={store}>
      <Services.Provider value={services}>
        <MessageProvider>
          <Content />
        </MessageProvider>
      </Services.Provider>
    </Provider>);

    const pageComponent = component.root.findByProps({id: 'main-content'});

    expect(pageComponent).toBeDefined();
  });

  it('renders with ToC in null state', () => {
    const component = renderer.create(<Provider store={store}>
      <Services.Provider value={services}>
        <MessageProvider>
          <Content />
        </MessageProvider>
      </Services.Provider>
    </Provider>);

    const sidebarComponent = component.root.findByType(Sidebar);

    expect(sidebarComponent.props.isOpen).toBe(null);
  });

  it('clicking overlay closes toc', () => {
    renderer.act(() => {
      store.dispatch(openToc());
    });

    const component = renderer.create(<Provider store={store}>
      <Services.Provider value={services}>
        <MessageProvider>
          <Content />
        </MessageProvider>
      </Services.Provider>
    </Provider>);

    const sidebarComponent = component.root.findByType(Sidebar);
    const mobileScrollLock = component.root.findByType(MobileScrollLock);

    expect(sidebarComponent.props.isOpen).toBe(true);
    renderer.act(() => {
      mobileScrollLock.props.onClick();
    });
    expect(sidebarComponent.props.isOpen).toBe(false);
  });

  it('SidebarControl opens and closes ToC', () => {
    const component = renderer.create(<Provider store={store}>
      <Services.Provider value={services}>
        <MessageProvider>
          <Content />
        </MessageProvider>
      </Services.Provider>
    </Provider>);

    expect(component.root.findByType(Sidebar).props.isOpen).toBe(null);

    renderer.act(() => {
      component.root
        .findByProps({'aria-label': 'Click to close the Table of Contents'})
        .props.onClick();
    });

    expect(component.root.findByType(Sidebar).props.isOpen).toBe(false);

    renderer.act(() => {
      component.root
        .findByProps({'aria-label': 'Click to open the Table of Contents'})
        .props.onClick();
    });

    expect(component.root.findByType(Sidebar).props.isOpen).toBe(true);
  });
});
