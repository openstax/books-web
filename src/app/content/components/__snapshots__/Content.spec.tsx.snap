// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`content matches snapshot 1`] = `
Array [
  .c0 {
  -webkit-clip: rect(1px,1px,1px,1px);
  clip: rect(1px,1px,1px,1px);
  margin: 0;
  position: absolute;
  left: 0;
  top: 0;
  height: 1px;
  width: 1px;
  overflow: hidden;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c0:focus {
  -webkit-clip: auto;
  clip: auto;
  height: auto;
  width: auto;
  z-index: 20;
}

<a
    className="c0"
    href="#main-content"
    onClick={[Function]}
  >
    Skip to Content
  </a>,
  .c6 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  margin: 3rem;
  padding: 0 3rem 0 3rem;
  font-size: 18px;
  font-family: "Noto Sans";
}

.c6 * {
  overflow: initial;
}

.c7 {
  font-family: "Roboto Condensed",serif;
  font-size: 19px;
  line-height: 29px;
}

.c1 {
  background-color: white;
  overflow-x: hidden;
  -webkit-transition: all 300ms;
  transition: all 300ms;
  width: 300px;
}

.c2 {
  position: fixed;
  top: 10px;
  height: 40px;
  width: 40px;
  background: none;
  padding: 0;
  border: none;
  z-index: 1;
  outline: none;
  -webkit-transition: all 300ms;
  transition: all 300ms;
  -webkit-transform: translateX(0px);
  -ms-transform: translateX(0px);
  transform: translateX(0px);
  left: 240px;
}

.c2 span {
  width: 100%;
  background: #000;
  padding-top: 10%;
  display: block;
  visibility: visible;
  opacity: 1;
  -webkit-transition: all 300ms;
  transition: all 300ms;
}

.c2 span:nth-child(1) {
  -webkit-transform: rotateZ(45deg);
  -ms-transform: rotateZ(45deg);
  transform: rotateZ(45deg);
}

.c2 span:nth-child(2) {
  opacity: 0;
}

.c2 span:nth-child(3) {
  -webkit-transform: rotateZ(-45deg);
  -ms-transform: rotateZ(-45deg);
  transform: rotateZ(-45deg);
  margin-top: -20%;
}

.c3 {
  position: fixed;
  left: 0;
  top: 0;
  bottom: 0;
  width: 300px;
  overflow-y: auto;
  font-size: 15px;
  padding: 10px;
  -webkit-transition: all 300ms;
  transition: all 300ms;
  -webkit-transform: translateX(0px);
  -ms-transform: translateX(0px);
  transform: translateX(0px);
}

.c3 ol {
  padding-inline-start: 10px;
}

.c3 > * {
  -webkit-transition-property: visibility;
  transition-property: visibility;
  -webkit-transition-delay: 0s;
  transition-delay: 0s;
  visibility: visible;
}

.c4 {
  list-style: none;
}

.c5 {
  list-style: none;
  overflow: visible;
  position: relative;
}

.c5:before {
  font-weight: bold;
  content: '>';
  position: absolute;
  right: 100%;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
}

<div
    className="c0"
  >
    <div
      className="c1"
    >
      <button
        aria-label="Click to close the Table of Contents"
        className="c2"
        onClick={[Function]}
        role="button"
      >
        <span />
        <span />
        <span />
      </button>
      <div
        className="c3"
      >
        <nav>
          <ol>
            <li
              className="c4"
            >
              <a
                href="#"
              >
                book details
              </a>
            </li>
          </ol>
        </nav>
        <div>
          <h2>
            Table of Contents
          </h2>
          <nav>
            <ol>
              <li
                aria-label="Current Page"
                className="c5"
              >
                <a
                  dangerouslySetInnerHTML={
                    Object {
                      "__html": "<span class=\\"os-text\\">Test Page 1</span>",
                    }
                  }
                  href="/books/book-slug-1/pages/Test-Page-1"
                  onClick={[Function]}
                />
              </li>
              <li
                className="c4"
              >
                <h3
                  dangerouslySetInnerHTML={
                    Object {
                      "__html": "<span class=\\"os-number\\">1</span><span class=\\"os-divider\\"> </span><span class=\\"os-text\\">Test Chapter 1</span>",
                    }
                  }
                />
                <nav>
                  <ol>
                    <li
                      className="c4"
                    >
                      <a
                        dangerouslySetInnerHTML={
                          Object {
                            "__html": "<span class=\\"os-text\\">Test Page 2</span>",
                          }
                        }
                        href="/books/book-slug-1/pages/1-Test-Page-2"
                        onClick={[Function]}
                      />
                    </li>
                  </ol>
                </nav>
              </li>
              <li
                className="c4"
              >
                <h3
                  dangerouslySetInnerHTML={
                    Object {
                      "__html": "<span class=\\"os-number\\">1</span><span class=\\"os-divider\\"> </span><span class=\\"os-text\\">Test Chapter 2</span>",
                    }
                  }
                />
                <nav>
                  <ol>
                    <li
                      className="c4"
                    >
                      <a
                        dangerouslySetInnerHTML={
                          Object {
                            "__html": "<span class=\\"os-text\\">Test Page 3</span>",
                          }
                        }
                        href="/books/book-slug-1/pages/1-Test-Page-3"
                        onClick={[Function]}
                      />
                    </li>
                  </ol>
                </nav>
              </li>
              <li
                className="c4"
              >
                <h3
                  dangerouslySetInnerHTML={
                    Object {
                      "__html": "<span class=\\"os-number\\">1</span><span class=\\"os-divider\\"> </span><span class=\\"os-text\\">Test Chapter 3</span>",
                    }
                  }
                />
                <nav>
                  <ol>
                    <li
                      className="c4"
                    >
                      <a
                        dangerouslySetInnerHTML={
                          Object {
                            "__html": "<span class=\\"os-text\\">Test Page 4</span>",
                          }
                        }
                        href="/books/book-slug-1/pages/1-Test-Page-4"
                        onClick={[Function]}
                      />
                    </li>
                  </ol>
                </nav>
              </li>
              <li
                className="c4"
              >
                <h3
                  dangerouslySetInnerHTML={
                    Object {
                      "__html": "<span class=\\"os-number\\">1</span><span class=\\"os-divider\\"> </span><span class=\\"os-text\\">Test Chapter 4</span>",
                    }
                  }
                />
                <nav>
                  <ol>
                    <li
                      className="c4"
                    >
                      <a
                        dangerouslySetInnerHTML={
                          Object {
                            "__html": "<span class=\\"os-text\\">Test Page 5</span>",
                          }
                        }
                        href="/books/book-slug-1/pages/1-Test-Page-5"
                        onClick={[Function]}
                      />
                    </li>
                  </ol>
                </nav>
              </li>
              <li
                className="c4"
              >
                <h3
                  dangerouslySetInnerHTML={
                    Object {
                      "__html": "<span class=\\"os-number\\">1</span><span class=\\"os-divider\\"> </span><span class=\\"os-text\\">Test Chapter 5</span>",
                    }
                  }
                />
                <nav>
                  <ol>
                    <li
                      className="c4"
                    >
                      <a
                        dangerouslySetInnerHTML={
                          Object {
                            "__html": "<span class=\\"os-text\\">Test Page 6</span>",
                          }
                        }
                        href="/books/book-slug-1/pages/1-Test-Page-6"
                        onClick={[Function]}
                      />
                    </li>
                  </ol>
                </nav>
              </li>
              <li
                className="c4"
              >
                <h3
                  dangerouslySetInnerHTML={
                    Object {
                      "__html": "<span class=\\"os-number\\">1</span><span class=\\"os-divider\\"> </span><span class=\\"os-text\\">Test Chapter 6</span>",
                    }
                  }
                />
                <nav>
                  <ol>
                    <li
                      className="c4"
                    >
                      <a
                        dangerouslySetInnerHTML={
                          Object {
                            "__html": "<span class=\\"os-text\\">Test Page 7</span>",
                          }
                        }
                        href="/books/book-slug-1/pages/1-Test-Page-7"
                        onClick={[Function]}
                      />
                    </li>
                  </ol>
                </nav>
              </li>
            </ol>
          </nav>
        </div>
      </div>
    </div>
    <div
      className="c6"
      id="main-content"
      tabIndex={-1}
    >
      <div>
        <h1
          className="c7"
        >
          Test Book 1
           / 
          Test Page 1
        </h1>
      </div>
      <div
        className=""
      >
        <div
          data-type="chapter"
        >
          <div
            dangerouslySetInnerHTML={
              Object {
                "__html": "<div xmlns:m=\\"http://www.w3.org/1998/Math/MathML\\" xmlns:ns2=\\"urn:oasis:names:tc:opendocument:xmlns:xsl-fo-compatible:1.0\\" class=\\"introduction\\" data-cnxml-to-html-ver=\\"1.3.2\\" data-type=\\"page\\" id=\\"1d1fd537-77fb-4eac-8a8a-60bbaa747b6d\\"><h2 data-type=\\"document-title\\"><span class=\\"os-number\\">1.1</span><span class=\\"os-divider\\"> </span><span class=\\"os-text\\">Test Page 1</span></h2><p id=\\"p1\\">Lorem ipsum dolor sit amet, consectetur adipiscing elit. Morbi ex erat, lacinia vitae mattis id, commodo vitae est. Nam quis arcu quis enim congue aliquet. Vivamus dictum rhoncus tortor, malesuada placerat tortor imperdiet ac. Fusce ac viverra nisi. Aliquam ut tempor diam. Cras rhoncus sodales massa vitae porttitor. Integer pharetra lorem a ante sollicitudin condimentum. Aliquam imperdiet erat et tellus mattis finibus. Sed nisl lectus, ultricies et ante ut, porttitor varius sem. Interdum et malesuada fames ac ante ipsum primis in faucibus. Fusce tempus interdum ante, nec placerat felis facilisis quis. Nunc nec malesuada nisi. Cras eu iaculis felis. Sed id maximus enim, sit amet sodales sapien. Proin pulvinar sem risus, nec ultricies odio viverra a. Aliquam feugiat euismod dapibus.</p><section id=\\"section1\\"><h3 data-type=\\"document-title\\" id=\\"100_copy_1\\"><span class=\\"os-number\\">1.1</span><span class=\\"os-divider\\"> </span><span class=\\"os-text\\">A Section, Probably</span></h3><p id=\\"p2\\">Morbi lobortis mattis velit, dapibus convallis est mollis sed. Sed ullamcorper est tortor, at ultricies felis tincidunt ut. Mauris interdum nunc et convallis pharetra. Nulla ac erat nulla. Vivamus a ornare leo. Quisque luctus dui eget auctor tristique. Sed vel est eget lorem volutpat mattis non et nisi. Aliquam ultricies ultrices velit, id fringilla ex suscipit in. Lorem ipsum dolor sit amet, consectetur adipiscing elit. Mauris consequat blandit felis, non finibus neque dignissim sit amet. Integer laoreet dapibus quam eget pulvinar. Aenean lobortis, arcu dignissim euismod ornare, eros est tincidunt urna, nec auctor odio quam eget tortor. Praesent sit amet metus a metus varius tincidunt. Pellentesque convallis sit amet erat eget malesuada. Proin pretium est euismod risus facilisis dignissim. Interdum et malesuada fames ac ante ipsum primis in faucibus.</p><p id=\\"p3\\">Integer maximus augue vitae orci vehicula tincidunt. Integer auctor ante odio, nec porta nisi pellentesque id. Donec porttitor velit vel turpis sagittis molestie. Sed eu rhoncus orci. Donec feugiat tristique posuere. Fusce consequat, nisi vitae porttitor ornare, augue velit sodales augue, ac facilisis turpis orci vitae lectus. Mauris mauris urna, dictum a dui et, imperdiet commodo justo. Pellentesque scelerisque dui sed libero porta, sit amet rhoncus ex dictum. Interdum et malesuada fames ac ante ipsum primis in faucibus.</p><p id=\\"p4\\">Donec congue tortor sed magna ullamcorper mollis. Suspendisse eu tincidunt augue, et pulvinar purus. Fusce gravida quam lorem, sed pharetra libero bibendum et. Etiam facilisis ex vel nisi egestas, non consequat mi malesuada. Donec scelerisque magna sem, ut ornare massa rhoncus sed. In ultricies tellus porttitor enim elementum, ut mollis nisl euismod. Pellentesque eu lobortis neque. Phasellus eget luctus velit, in eleifend lectus. Pellentesque leo odio, cursus eget feugiat eu, finibus in nibh. In nec urna justo. Donec ut nisi eleifend, auctor lacus quis, hendrerit sem. Pellentesque sagittis nunc a molestie finibus. Donec pulvinar nibh elit, nec ornare nibh egestas quis. Curabitur luctus euismod lobortis.</p><p id=\\"p5\\">Maecenas vehicula nec quam vitae sodales. Vestibulum bibendum accumsan mauris. Nunc rhoncus libero odio, in ullamcorper massa ornare convallis. Nullam metus tortor, aliquam a ultrices et, rhoncus in risus. Nunc vel turpis erat. Cras semper sagittis odio, eu mattis nulla faucibus quis. Aliquam egestas nunc elit, eget condimentum lectus congue et. Curabitur non gravida lorem. Integer mi nunc, commodo vitae augue vel, bibendum pretium magna. Sed ut nibh venenatis ipsum tempus mollis pretium eget est. Aliquam commodo quam ipsum, at fermentum enim tristique eget. Mauris imperdiet dapibus maximus. Donec tristique varius lacinia. Pellentesque pharetra nunc eu nisi ullamcorper placerat. Suspendisse potenti.</p><p id=\\"p6\\">Integer fringilla blandit dolor ut mattis. Nam commodo ex at blandit sodales. Nullam suscipit, nunc eu mollis sagittis, purus arcu tincidunt ligula, vitae pharetra metus tortor id ex. Donec malesuada aliquet ligula id mollis. Maecenas scelerisque maximus sollicitudin. Quisque sagittis dolor at ligula congue, quis rhoncus mi porttitor. Quisque lobortis ut ante sed convallis. Integer luctus cursus molestie.</p><p id=\\"p7\\">Morbi a purus a elit dapibus fermentum. Sed varius ex quis tortor tincidunt posuere. Mauris luctus risus id aliquet mollis. Phasellus ornare leo a arcu imperdiet dictum. Praesent accumsan venenatis urna, sed egestas sem pharetra a. Sed vulputate sit amet nisl a aliquet. Interdum et malesuada fames ac ante ipsum primis in faucibus. Mauris facilisis augue vitae est maximus pharetra in ac elit. Pellentesque viverra purus non ligula tempor convallis.</p><p id=\\"p8\\">Pellentesque lectus ante, malesuada ut ornare vitae, euismod non urna. Morbi varius eros eget felis accumsan vehicula. In diam ex, finibus rhoncus dapibus non, lacinia in lacus. Nullam pretium mollis laoreet. Ut interdum dui sit amet elementum dignissim. Etiam in nibh vitae ligula fringilla aliquet quis vel ex. Nullam quis elit libero. In aliquet laoreet arcu at posuere. Duis consectetur et dolor non accumsan. Fusce enim est, dignissim sed mauris at, ornare cursus ligula. Nullam tincidunt eu purus vel lacinia. Praesent varius, arcu sed facilisis elementum, orci metus vestibulum nisl, sed sodales urna eros ullamcorper turpis. Donec mollis ac massa ac elementum. Nulla maximus purus sed orci efficitur, ut porta mauris elementum.</p><p id=\\"p9\\">Vestibulum blandit enim quis sapien ultrices pretium. Cras dapibus ornare fringilla. Duis ut laoreet dolor. Proin ullamcorper molestie dui eu vestibulum. Nullam convallis mollis mauris nec accumsan. Proin egestas enim non nisl suscipit, ut feugiat felis pharetra. In faucibus fringilla ipsum vel sollicitudin. Maecenas blandit augue odio, ac mattis justo feugiat sit amet. Etiam nec molestie eros. Sed gravida velit libero, at pretium diam venenatis sit amet. Donec tellus leo, faucibus eu sapien nec, mollis efficitur lorem. Lorem ipsum dolor sit amet, consectetur adipiscing elit.</p><p id=\\"p10\\">Mauris molestie diam a felis feugiat mattis. Nulla sit amet dui quis ante lacinia iaculis. Suspendisse gravida ligula a pharetra pellentesque. Proin et augue eu nunc tincidunt euismod a vitae est. Mauris fringilla sagittis massa at molestie. Mauris pharetra condimentum aliquam. Maecenas a velit vitae nisi placerat varius id in mauris. Sed a nunc vehicula, vestibulum velit vitae, ullamcorper ante. Praesent elementum euismod bibendum. Phasellus blandit ex id sem porta, vel scelerisque nibh finibus. Etiam ultrices sollicitudin nunc, et dictum orci lacinia eget. Cras at nulla placerat elit varius aliquam. In vitae sapien interdum, semper mauris vel, volutpat metus. Pellentesque vel nisl vel est dictum fringilla. Etiam fermentum, lorem in sodales vehicula, dui tellus lobortis orci, nec aliquet erat eros quis magna. Quisque scelerisque convallis malesuada.</section></div>",
              }
            }
            data-type="page"
          />
        </div>
      </div>
    </div>
  </div>,
]
`;

exports[`content renders empty state 1`] = `
Array [
  .c0 {
  -webkit-clip: rect(1px,1px,1px,1px);
  clip: rect(1px,1px,1px,1px);
  margin: 0;
  position: absolute;
  left: 0;
  top: 0;
  height: 1px;
  width: 1px;
  overflow: hidden;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c0:focus {
  -webkit-clip: auto;
  clip: auto;
  height: auto;
  width: auto;
  z-index: 20;
}

<a
    className="c0"
    href="#main-content"
    onClick={[Function]}
  >
    Skip to Content
  </a>,
  .c5 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  margin: 3rem;
  padding: 0 3rem 0 3rem;
  font-size: 18px;
  font-family: "Noto Sans";
}

.c5 * {
  overflow: initial;
}

.c1 {
  background-color: white;
  overflow-x: hidden;
  -webkit-transition: all 300ms;
  transition: all 300ms;
  width: 300px;
}

.c2 {
  position: fixed;
  top: 10px;
  height: 40px;
  width: 40px;
  background: none;
  padding: 0;
  border: none;
  z-index: 1;
  outline: none;
  -webkit-transition: all 300ms;
  transition: all 300ms;
  -webkit-transform: translateX(0px);
  -ms-transform: translateX(0px);
  transform: translateX(0px);
  left: 240px;
}

.c2 span {
  width: 100%;
  background: #000;
  padding-top: 10%;
  display: block;
  visibility: visible;
  opacity: 1;
  -webkit-transition: all 300ms;
  transition: all 300ms;
}

.c2 span:nth-child(1) {
  -webkit-transform: rotateZ(45deg);
  -ms-transform: rotateZ(45deg);
  transform: rotateZ(45deg);
}

.c2 span:nth-child(2) {
  opacity: 0;
}

.c2 span:nth-child(3) {
  -webkit-transform: rotateZ(-45deg);
  -ms-transform: rotateZ(-45deg);
  transform: rotateZ(-45deg);
  margin-top: -20%;
}

.c3 {
  position: fixed;
  left: 0;
  top: 0;
  bottom: 0;
  width: 300px;
  overflow-y: auto;
  font-size: 15px;
  padding: 10px;
  -webkit-transition: all 300ms;
  transition: all 300ms;
  -webkit-transform: translateX(0px);
  -ms-transform: translateX(0px);
  transform: translateX(0px);
}

.c3 ol {
  padding-inline-start: 10px;
}

.c3 > * {
  -webkit-transition-property: visibility;
  transition-property: visibility;
  -webkit-transition-delay: 0s;
  transition-delay: 0s;
  visibility: visible;
}

.c4 {
  list-style: none;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
}

<div
    className="c0"
  >
    <div
      className="c1"
    >
      <button
        aria-label="Click to close the Table of Contents"
        className="c2"
        onClick={[Function]}
        role="button"
      >
        <span />
        <span />
        <span />
      </button>
      <div
        className="c3"
      >
        <nav>
          <ol>
            <li
              className="c4"
            >
              <a
                href="#"
              >
                book details
              </a>
            </li>
          </ol>
        </nav>
      </div>
    </div>
    <div
      className="c5"
      id="main-content"
      tabIndex={-1}
    >
      <div />
      <div
        className=""
      >
        <div
          data-type="chapter"
        >
          <div
            dangerouslySetInnerHTML={
              Object {
                "__html": "",
              }
            }
            data-type="page"
          />
        </div>
      </div>
    </div>
  </div>,
]
`;
